### ISSUE TITLE
508-defect-1❗[SCREENREADER]: Feedback Form - Satisfaction radio button labels not read by NVDA in Browse mode

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#26563]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a screen reader user, I want to hear unique labels read out for each radio button. These should be announced consistently in three places:
1. The Forms menu
2. When navigating text nodes in Browse mode
3. When entering data in Forms/Focus mode

## Details

The five radio buttons for Overall Satisfaction are not being read by NVDA + Firefox in Browse mode. This is one of the three main ways to navigate a form, and could cause a high level of frustration. Based on the latest WebAIM poll, NVDA is the [most popular screen reader](https://webaim.org/projects/screenreadersurvey8/#primary) among respondents in the 2019 survey.

Screenshot attached below.

## Acceptance Criteria

- [x] All five radio buttons have a unique label that includes a number and text description. IE "2 - Somewhat dissatisfied"
- [ ] Valid markup with 0 axe-core errors
- [ ] ~Ideally these are formatted vertically and use the same basic radio button visual styling as the first group of radio buttons on the form~

## WCAG or Vendor Guidance (optional)

* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

## Screenshots or Trace Logs

![Screen Shot 2021-06-24 at 10 46 47 AM](https://user-images.githubusercontent.com/934879/123294925-e41d0c00-d4da-11eb-82f8-ca368f40416c.png)

**Comment by holdenhinkle** [comment]
I’ve poked around with Medallia a bunch, learned how to use it, recreated the form that's in this ticket, looked at the css, figured out some possible css overrides, made lots of notes.

Google doc with notes - https://docs.google.com/document/d/18_wEnSoLWrWfVWuOVOd3gcS50McgG3AjLtZ85FKAFaM/edit?usp=sharing

I'm meeting with Trevor tomorrow morning to review the Medallia tickets.

**Comment by holdenhinkle** [comment]
The text for the satisfaction labels for this form is missing in Medallia:

![image](https://user-images.githubusercontent.com/9326247/125488229-de77973c-439e-48c9-a4c2-260e4bf5bd48.png)

Text needs to be entered for options 2, 3, and 4.

Proposed text:
2 - Dissatisfied
3 - Okay
4 - Satisfied

I'm trying to find out if it's okay to go ahead and make the change, or if someone else needs to, or if there's an approval process I need to go through before I make the change.

**Comment by holdenhinkle** [comment]
I reviewed this ticket with Trevor and showed him how the rating scale is being rendered (we looked at the markup together). He said that know that he knows how it works, I should just update the missing values for the labels (see my previous post above) and that I should not make the other changes (css overrides that vertically stack the blue radio rating buttons, make the rating buttons look like the yes/no radio buttons above them, etc.).

**Comment by jwoodman5** [comment]
@holdenhinkle Based on working use elsewhere on VA.gov survey, the proposed text for #2 and #4 looks good. For #3, let's use `Neither dissatisfied nor satisfied`.

**Comment by holdenhinkle** [comment]
I chatted with Jason. He gave me approval to update the missing values for the `aria` labels - 

![image](https://user-images.githubusercontent.com/9326247/125495554-9b2b7717-b979-4fd3-b662-43e98c674f89.png)

I'll do that after I have lunch after the VSP Retro.

**Comment by holdenhinkle** [comment]
I updated the rating scales for both the `VA.gov VFS-Search Questionnaire` and `VA.gov VFSQuestionnaire` forms in both staging and production to the following:

![image](https://user-images.githubusercontent.com/9326247/125501431-6dbb3618-88cf-46d3-ba8c-0a9a28ebf675.png)

The missing labels are present now in staging and production. 

Pic of VA.gov (production):

<img width="1298" alt="2021-07-13_13-58-40" src="https://user-images.githubusercontent.com/9326247/125502348-c0fe6672-a838-44ba-8c13-9336872f3a7d.png">

**Comment by holdenhinkle** [comment]
I tested [this page](https://www.va.gov/health-care/refill-track-prescriptions/) with the Firefox Axe DevTools plugin and I don’t see any errors related to the form.

![image](https://user-images.githubusercontent.com/9326247/125504057-8e791185-ae30-4210-a90b-1aeee1443e44.png)

![image](https://user-images.githubusercontent.com/9326247/125504230-3843d44e-b9db-43db-a887-cd2bbc1de515.png)

@1Copenut, please let me know if you get any errors with NVDA.

**Comment by holdenhinkle** [comment]
trevorpierce  2:25 PM
I just checked this in FF + NVDA on the linked page. FF still doesn't read out the labels in browse mode, but does have them listed properly in the Forms menu and reads them accurately in Focus mode (typical for form data entry).

2:25
I think we have to put this one in the win column

Holden Hinkle  2:25 PM
Okay, awesome :+1:
2:26
Thanks for checking

This ticket is done.

**Comment by jwoodman5** [comment]
@holdenhinkle I see this one is "done". Do you think we can close it out? Thanks.

**Comment by holdenhinkle** [comment]
I always put my tickets into Review/QA when I'm done with them. @pjhill , want to close this out?

**Comment by pjhill** [comment]
I verified the fix implemented is live in production as well. Closing this ticket!
